src/http/server.clj

```diff
@@ -325,6 +334,34 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn password-prompt! []
+  (if-let [console (System/console)]
+    (let [pw (String. (.readPassword console  "Enter password: " nil))]
+      (if (empty? pw)
+        (do (println "Password cannot be empty!")
+            (recur))
+        pw))
+    (throw (Exception. "No console available"))))
+
+(defn maybe-inject-auth
+  [auth handler]
+  (if (some? auth)
+    (fn [{:keys [headers] :as req}]
+      (if (not= (headers "Authorization") auth)
+        (handler (-> (update req :headers dissoc "authorization")
```

Comment:
You don't need to call a handler here. Just return an error response and we're done.